io/io/src/TStreamerInfoActions.cxx

```diff
@@ -45,6 +45,12 @@ using namespace TStreamerInfoActions;
 
 namespace TStreamerInfoActions
 {
+   bool IsDefaultVector(TVirtualCollectionProxy &proxy)
+   {
+      return ((proxy.GetCollectionType() == ROOT::kSTLvector && !(proxy.GetProperties() & TVirtualCollectionProxy::kCustomAlloc))
+              || (proxy.GetProperties() & TVirtualCollectionProxy::kIsEmulated) );
```

Comment:
Is this more readable?
```
if (proxy.GetCollectionType() == ROOT::kSTLvector) {
   int props = proxy.GetProperties();
   if (!(props & TVirtualCollectionProxy::kCustomAlloc)) {
      return !(props & TVirtualCollectionProxy::kIsEmulated);
   }
}
return false;
```

Reply:
humm ... maybe more readable but also incorrect/different :)

Reply:
Yeah, I meant *after* applying C++ auto-correction, of course ;-)

Reply:
The auto-correct makes the change not as appeal though
```
auto props = proxy.GetProperties();
if (proxy.GetCollectionType() == ROOT::kSTLvector) {
   return  !(props & TVirtualCollectionProxy::kCustomAlloc);
} else {
   return (props & TVirtualCollectionProxy::kIsEmulated);
}
```
or
```
auto props = proxy.GetProperties();
return ((proxy.GetCollectionType() == ROOT::kSTLvector && !(props & TVirtualCollectionProxy::kCustomAlloc))
        || (props & TVirtualCollectionProxy::kIsEmulated) );
```
or
```
auto props = proxy.GetProperties();
bool isVector = proxy.GetCollectionType() == ROOT::kSTLvector;
bool hasDefaultAlloc = !(props & TVirtualCollectionProxy::kCustomAlloc);
bool isEmulated = TVirtualCollectionProxy::kIsEmulated;

return isEmulated || (isVector && hasDefaultAlloc);